src/visualizers/VisualizerChaos.ts

```diff
@@ -0,0 +1,393 @@
+			}
+			colorList.push( hexString ); 
+		}
+		this.randomPalette = new Palette(this.sketch, colorList, this.darkMode ? '#262626' : '#f5f5f5', this.darkMode ? '#f5f5f5' : '#262626' );
+
+		// the second and third palette are curated
+		this.darkPalette = new Palette(this.sketch,['#48458b','#daa520','#003f5c','#ff6361','#ffa600','#bc5090','#58508d'],'#262626','#f5f5f5');
+		this.lightPalette = new Palette(this.sketch,['#48458b','#daa520','#003f5c','#ff6361','#ffa600','#bc5090','#58508d'],'#f5f5f5','#262626');
+		
+		// decide which palette to set by default
+		// we need a colourpicker in the params eventually
+		// right now this is a little arbitrary
+		if( this.darkMode ){ 
+			this.currentPalette = this.darkPalette;
+		} else {
+			this.currentPalette = this.lightPalette;
+		}
+		if( 
+			(this.colorStyle === ColorStyle.Walker && this.walkers > 7) 
+			|| ( this.colorStyle === ColorStyle.Corner && this.corners > 7 ) 
+		){ this.currentPalette = this.randomPalette; } 
+		
+		// set center coords and size
+		const center = this.sketch.createVector( this.sketch.width * 0.5, this.sketch.height * 0.5 );
+		const radius = this.sketch.width * 0.4;
+
+		// text appearance control
+		const labelOutset = 1.1;
+		const shrink = Math.log(this.corners);
+		const textSize = this.sketch.width * 0.04 / shrink; // shrinks the numbers appropriately up to about 100 corners or so
+		const textStroke = this.sketch.width * 0; // no stroke right now, but could be added
+
+		// set counters
+		this.pixelCount = 0;
+		this.myIndex = 0;
+
+		// set up arrays of walkers
+		this.walkerPositions.splice(0, this.walkerPositions.length) // clean the array out (fixes bug with redraws)
+		for( let w = 0; w < this.walkers; w++ ){
+			this.walkerPositions.push(center); // all walkers start at origin
+		}
+
+		// Set up the windows and return the coordinates of the corners
+		this.cornersList = this.chaosWindow( center, radius); // locations of the corners
+
+		// Set frame rate
+		this.sketch.frameRate(10);
+
+		// canvas clear/background
+		this.sketch.clear();
+		this.sketch.background(this.currentPalette.backgroundColor);
+
+		// Draw corner labels if desired
+		if ( this.showLabels ) {
+			this.sketch.stroke(this.currentPalette.textColor);
+			this.sketch.fill(this.currentPalette.textColor);
+			this.sketch.strokeWeight(textStroke);
+			this.sketch.textSize(textSize);
+			this.sketch.textAlign(this.sketch.CENTER,this.sketch.CENTER);
+			const cornersLabels = this.chaosWindow(center,(radius)*(labelOutset)); // locations of the labels
+			for ( let c = 0 ; c < this.corners ; c++ ) {
+				const label = cornersLabels[c];
+				this.sketch.text(String(c),label.x,label.y);
+			}
+		}
+
+		// no stroke (in particular, no outline on circles)
+		this.sketch.strokeWeight(0);
+	}
+
+	draw() {
+
+		super.draw();
+
+		// we do pixelsPerFrame pixels each time through the draw cycle; this speeds things up essentially
+		for( let px = 0; px < this.pixelsPerFrame; px++ ){
+			
```

Comment:
You could probably just use the existing myIndex or pixelCount or the unified version thereof if you go that way as the loop index, with something like:
```
const pixelLimit = this.pixelCount + this.pixelsPerFrame;
for (; this.pixelCount < pixelLimit; this.pixelCount++) {
  ... blah blah..
```
This is mostly a matter of taste; I tend to prefer creating fewer variables, all else being equal. Another possible argument in favor of pre-calculating the limit and using pixelCount as the loop variable is that you might be able to do something like
`const pixelLimit = min(this.num, this.pixelCount + this.pixelsPerFrame);`
and thereby simplify some of the termination logic in the loop -- possibly; this comment is just food for thought.